tests/ballerina-test-integration/src/test/java/org/ballerinalang/test/packaging/PackagingPushTestCase.java

```diff
@@ -0,0 +1,266 @@
+                                                      "functions.bal").getAbsolutePath());
+        Files.copy(srcFilePath, src.resolve("functions.bal"));
+
+        Path obj = tempDir.resolve("obj");
+        Files.createDirectories(obj);
+        Path objFilePath = Paths.get(new File("src" + File.separator + "test" + File.separator + "resources"
+                                                      + File.separator + "packaging" + File.separator +
+                                                      "my.app.balo").getAbsolutePath());
+        Files.copy(objFilePath, obj.resolve(packageName + ".balo"));
+
+        Path mdDir = tempDir.resolve(packageName);
+        Files.createDirectories(mdDir);
+        Path mdFilePath = Paths.get(new File("src" + File.separator + "test" + File.separator + "resources"
+                                                     + File.separator + "packaging" + File.separator +
+                                                     "Package.md").getAbsolutePath());
+        Files.copy(mdFilePath, mdDir.resolve("Package.md"));
+
+        compressFiles(tempDir, new FileOutputStream(projectPath.resolve(generatedPackagePath)
+                                                               .resolve(packageName + ".zip").toFile()));
+    }
+
+    @Test(description = "Test pushing a package to central")
+    public void testPush() throws Exception {
+        ballerinaClient = new ServerInstance(serverZipPath);
+        String sourceRootPath = projectPath.toString();
+        String[] clientArgs = {"--sourceroot", sourceRootPath, packageName};
+
+        String msg = "integrationtests/" + packageName + ":1.0.0 [project repo -> central]";
+
+        LogLeecher clientLeecher = new LogLeecher(msg);
+        ballerinaClient.addLogLeecher(clientLeecher);
+        ballerinaClient.runMain(clientArgs, getEnvVariables(), "push");
+        clientLeecher.waitForText(5000);
+
+        // Check if package Exists
+        String stagingURL = "https://api.staging-central.ballerina.io/packages/";
+        HttpResponse response = HttpClientRequest.doGet(stagingURL + "integrationtests/" + packageName
+                                                                + "/1.0.0");
+        Assert.assertEquals(response.getResponseCode(), 200, "Response code mismatched");
+
+    }
+
+    /**
+     * Get environment variables and add ballerina_home as a env variable the tmp directory.
+     *
+     * @return env directory variable array
+     */
+    private String[] getEnvVariables() {
+        List<String> variables = new ArrayList<>();
+
+        Map<String, String> envVarMap = System.getenv();
+        envVarMap.forEach((key, value) -> variables.add(key + "=" + value));
+        variables.add(ProjectDirConstants.HOME_REPO_ENV_KEY + "=" + tempHomeDirectory.toString());
+        variables.add("BALLERINA_DEV_STAGE_CENTRAL" + "=" + "true");
+
+        return variables.toArray(new String[0]);
```

Comment:
toArray shout get a array of correct length.

Reply:
Fixed with 2f3e132

https://docs.oracle.com/javase/7/docs/api/java/util/Collection.html#toArray%28T%5B%5D%29